src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -165,8 +169,20 @@ Transport getTransport() {
         this.callback = (payload) -> {
             resetServerTimeout();
             if (!handshakeReceived) {
-                int handshakeLength = payload.indexOf(RECORD_SEPARATOR) + 1;
-                String handshakeResponseString = payload.substring(0, handshakeLength - 1);
+                List<Byte> handshakeByteList = new ArrayList<Byte>();
```

Comment:
There's a tradeoff there - if the ByteBuffer is readOnly, I'll have the following up to above this block, rather than moving it to MessagePackHubProtocol: https://github.com/dotnet/aspnetcore/blob/48d0b9f5d8b211043262d06ac7c08e1fc43c6b2f/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java#L210-L215